app/celery/scheduled_tasks.py

```diff
@@ -189,3 +199,140 @@ def check_templated_letter_state():
                 message=msg,
                 ticket_type=zendesk_client.TYPE_INCIDENT
             )
+
+
+def _get_dynamodb_comp_pen_messages(table, message_limit: int) -> list:
+    """
+    Helper function to get the Comp and Pen data from our dynamodb cache table.
+
+    :param table: the dynamodb table to grab the data from
+    :param message_limit: the number of rows to search at a time and the max number of items that should be returned
+    :return: a list of entries from the table that have not been processed yet
+    """
+
+    results = table.scan(
+        FilterExpression=boto3.dynamodb.conditions.Attr('is_processed').eq(False),
+        Limit=message_limit
+    )
+
+    items: list = results.get('Items')
+
+    # Keep getting items from the table until we have the number we want to send, or run out of items
+    while 'LastEvaluatedKey' in results and len(items) < message_limit:
+        results = table.scan(
+            FilterExpression=boto3.dynamodb.conditions.Attr('is_processed').eq(False),
+            Limit=message_limit,
+            ExclusiveStartKey=results['LastEvaluatedKey']
+        )
+
+        items.extend(results['Items'])
```

Comment:
I believe this causes an infinite loop if results['Items'] is an empty list.

Reply:
It will always end when we run out of things to look for in the dynamodb table, so I don't think there's anything to worry about there.

Reply:
I think `test_get_dynamodb_comp_pen_messages_with_data` would test that case. If we upgrade and they break that functionality we'll see it when testing. No worries then!